app/bundles/LeadBundle/Form/Type/CompanyType.php

```diff
@@ -90,32 +90,40 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 FormButtonsType::class
             );
         }
-        $builder->add(
-            'buttons',
-            FormButtonsType::class,
-            [
-                'post_extra_buttons' => [
-                    [
-                        'name'  => 'merge',
-                        'label' => 'mautic.lead.merge',
-                        'attr'  => [
-                            'class'       => 'btn btn-default btn-dnd',
-                            'icon'        => 'ri-exchange-2-line',
-                            'data-toggle' => 'ajaxmodal',
-                            'data-target' => '#MauticSharedModal',
-                            'data-header' => $this->translator->trans('mautic.lead.company.header.merge'),
-                            'href'        => $this->router->generate(
-                                'mautic_company_action',
-                                [
-                                    'objectId'     => $options['data']->getId(),
-                                    'objectAction' => 'merge',
-                                ]
-                            ),
+
+        if (null === $options['data']->id) {
```

Comment:
Are you sure this will work? On the line 117 I can see it's accessed as `$options['data']->getId()`

Reply:
It did work when I logged it, I also checked for the expected behavior and it worked fine. Should I change it to `getId()` to be on the safer side?

Reply:
There is some Twig magic where it looks if the property exists, if not it tries if `getId()` exists and if so it will use that. I'd prefer to use directly the entity method than the magic.
```suggestion
        if (null === $options['data']->getId()) {
```